core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -252,16 +283,66 @@ class CityConstructions {
             cancelCurrentConstruction()
     }
 
-    fun cancelCurrentConstruction(){
-        currentConstructionIsUserSet=false
-        currentConstruction=""
+    private fun cancelCurrentConstruction() {
+        currentConstructionIsUserSet = false
+        currentConstruction = ""
         chooseNextConstruction()
     }
 
     fun chooseNextConstruction() {
         if(currentConstructionIsUserSet) return
+
+        if (constructionQueue.isNotEmpty()) {
+
+            currentConstructionIsUserSet = true
+            currentConstruction = constructionQueue.removeAt(0)
+            stopUnbuildableConstruction()
+
+            if (currentConstruction != "") return
+        }
+
         ConstructionAutomation(this).chooseNextConstruction()
     }
-    //endregion
 
+    fun isEnqueuable(constructionName: String): Boolean {
+        return true
+    }
```

Comment:
is this ok? looks like useless function

Reply:
Oops, I'll clean this. It isn't needed in last implementation